### PR TITLE
[4.0] Move csp configuration to the com_csp component

### PR DESCRIPTION
**Pull request by zero-24**
### Summary of Changes

Move csp configuration to the com_csp component

### Testing Instructions

configure the http headers using com_csp's options make sure all is still working as expected

### Expected result

csp configs are just moved and all features still work

### Actual result

csp config is in the plugin but we now have a component so the options get moved

### Documentation Changes Required

none

cc @SniperSister

### CODE REVIEW

**Comment by brianteeman** [comment]
I think that instead of having the comments in the language strings we should adopt the method used in the installer and use sprintf so the strings dont contain the links and terms that should not be translated

**Comment by zero-24** [comment]
Agree but IIRC there is no way to do this for strings in the XML files right?

**Comment by brianteeman** [comment]
Sorry I forgot these were in the xml

File: administrator/components/com_csp/config.xml [hunk 1]
```diff
@@ -1,5 +1,87 @@
...
+			multiple="true"
+			min="1"
+			showon="contentsecuritypolicy:1[AND]contentsecuritypolicy_mode:custom"
+		>
+			<form>
+				<field
+					name="directive"
+					type="text"
+					label="COM_CSP_CONTENTSECURITYPOLICY_VALUES_DIRECTIVE"
+					class="span4"
```
Comment (Quy) [inline administrator/components/com_csp/config.xml]:
```suggestion
					class="col-md-4"
```

File: administrator/components/com_csp/config.xml [hunk 2]
```diff
@@ -1,5 +1,87 @@
...
+				/>
+				<field
+					name="value"
+					type="text"
+					label="COM_CSP_CONTENTSECURITYPOLICY_VALUES_VALUE"
+					class="span10"
```
Comment (Quy) [inline administrator/components/com_csp/config.xml]:
```suggestion
					class="col-md-10"
```

File: administrator/components/com_csp/config.xml [hunk 3]
```diff
@@ -1,5 +1,87 @@
...
+				/>
+				<field
+					name="client"
+					type="radio"
+					label="COM_CSP_HEADER_CLIENT"
+					default="site"
+					class="span12"
```
Comment (Quy) [inline administrator/components/com_csp/config.xml]:
```suggestion
					class="col-md-12"
```

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**